File: python/128_Longest_Consecutive_Sequence.py
class Longest_Consecutive_Sequence:
  def longestConsecutive(self, nums: list[int]) -> int:
    myNums = set(nums)
    longest = 0
    for n in nums:
      if (n-1) not in myNums:
        # We found the beginning of the sequence
        length = 0
        while ((n + length) in myNums):
          length += 1
        longest = max(length, longest)
    return longest

File: c++/128_Longest_Consecutive_Sequence.cpp
#include <set>
#include <vector>

using namespace std;

class Solution {
public:
  int longestConsecutive(vector<int>& nums) {
    set<int> myNums(nums.begin(), nums.end());
    if (myNums.empty()){
      return 0;
    }
    int length;
    int longest = 0;
    for (auto it = myNums.begin(); it != myNums.end(); it++){
      if (myNums.find(*it - 1) == myNums.end()){
        length = 0;
        while (myNums.find(*it + length) != myNums.end()){
          length++;
        }
        longest = max(length, longest);
        it = myNums.find(*it + length - 1);
      }
    }
    return longest;
  }
};